radicale/storage.py

```diff
@@ -207,19 +237,42 @@ def __init__(self, path):
 
 
 class Item:
-    def __init__(self, collection, item, href, last_modified=None):
+    def __init__(self, collection, href, last_modified=None,
+                 text=None, item=None, etag=None):
+        if text is None and item is None:
```

Comment:
``text`` is just the raw data from the storage. It will fail as soon as it's parsed with **vobject**, like any other invalid data. I don't think that we should introduce a special case for one kind of invalid data. That would also mean, that we have to duplicate the logging.